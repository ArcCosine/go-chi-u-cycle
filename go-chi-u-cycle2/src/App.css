.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.carousel-container {
  margin: 200px auto 0;
  perspective: 1000px;
  width: 250px;
  height: 250px;
  position: relative;
}

.rotate-container {
  margin: 40px auto;
}

.carousel-container .carousel {
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  transform-style: preserve-3d;
  transition: transform 1s;
}

.carousel figure {
  width: 250px;
  height: 250px;
  position: absolute;
}

.carousel figure img {
  width: 100%;
}
