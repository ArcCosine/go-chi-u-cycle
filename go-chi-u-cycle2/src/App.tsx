import './App.css'
import RotateButton from "./components/RotateButton";

function App() {

  return (
    <div className="App">
      <header className="App-header">
        <p>ごちうサイクル - ご注文はサイクルですか？ -</p>
      </header>
      <RotateButton />
    </div>
  )
}

export default App
