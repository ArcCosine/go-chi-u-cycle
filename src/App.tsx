import "./App.css";
import RotateButton from "./components/RotateButton";
import CycleArea from "./components/CycleArea";
import { CountProvider } from "./context/CountContext";

const images = [
    {
        imageUrl: "chino.jpg",
        imageText: "chino",
    },
    {
        imageUrl: "chino.png",
        imageText: "chino",
    },
    {
        imageUrl: "chino_anime.jpg",
        imageText: "chino_anime",
    },
    {
        imageUrl: "chiya.jpg",
        imageText: "chiya",
    },
    {
        imageUrl: "chiya.png",
        imageText: "chiya",
    },
    {
        imageUrl: "chiya_anime.jpg",
        imageText: "chino_anime",
    },
    {
        imageUrl: "cocoa.jpg",
        imageText: "cocoa",
    },
    {
        imageUrl: "cocoa.png",
        imageText: "cocoa",
    },
    {
        imageUrl: "cocoa_anime.jpg",
        imageText: "cocoa_anime",
    },
    {
        imageUrl: "rize.jpg",
        imageText: "rize",
    },
    {
        imageUrl: "rize.png",
        imageText: "rize",
    },
    {
        imageUrl: "rize_anime.jpg",
        imageText: "rize_anime",
    },
    {
        imageUrl: "sharo_anime.jpg",
        imageText: "sharo_anime",
    },
    {
        imageUrl: "syaro.jpg",
        imageText: "syaro",
    },
    {
        imageUrl: "syaro.png",
        imageText: "syaro",
    },
];

function App() {
    return (
        <div className="App">
            <header className="App-header">
                <h1>ごちうサイクル - ご注文はサイクルですか？ -</h1>
            </header>
            <CountProvider>
                <CycleArea images={images} />
                <RotateButton length={images.length} />
            </CountProvider>
            <footer>
                <div className="column-container">
                    <div className="column">
                        <h1>Special Thanks</h1>
                        <ul>
                            <li>
                                
                                <a href="https://gochiusa.com/bloom/">
                                    ご注文はうさぎですか？
                                </a>
                            </li>
                            <li>
                                <a href="https://gochiusa.com/special/c00370000.html">
                                    Twitterアイコン＆待ち受け配布
                                </a>
                            </li>
                            <li>
                                <a href="https://twitter.com/usagi_anime">
                                    TVアニメ『ご注文はうさぎですか？』公式Twitter
                                </a>
                            </li>
                        </ul>
                    </div>
                    <div className="column">
                        <h1>Powerd By</h1>
                        <ul>
                            <li>
                                <a href="https://vitejs.dev/">Vite</a>
                            </li>
                            <li>
                                <a href="https://reactjs.org/">React</a>
                            </li>
                            <li>
                                <a href="https://www.typescriptlang.org/">
                                    TypeScript
                                </a>
                            </li>
                        </ul>
                        <h1>Source</h1>
                        <ul>
                            <li>
                                <a href="https://github.com/ArcCosine/go-chi-u-cycle" target="_blank" rel="noopener">GitHub</a>
                            </li>
                        </ul>
                    </div>
                </div>
                <div className="copy-right">
                    Since:2016-2022 ごちうサイクル - ご注文はサイクルですか？ -{" "}
                    <a href="https://twitter.com/ArcCosine">&#64;ArcCosine</a>
                </div>
            </footer>
        </div>
    );
}

export default App;
