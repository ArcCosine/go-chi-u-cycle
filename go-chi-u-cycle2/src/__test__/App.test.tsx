import '@testing-library/jest-dom';
import { describe, expect, test } from "vitest";
import { render, screen } from "@testing-library/react";

import App from "../App";

describe("App dom test", () => {
    test("render test", () => {
        const { container } = render(<App />);
        expect(container.firstChild).toMatchSnapshot();
    });

    test("title text", () => {
        render(<App />);
        expect(screen.getByText(/ごちうサイクル - ご注文はサイクルですか？ -/i)).toBeInTheDocument();
    });
});
