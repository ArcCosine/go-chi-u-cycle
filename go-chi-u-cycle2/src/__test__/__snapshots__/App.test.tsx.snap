// Vitest Snapshot v1

exports[`App dom test > render test 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <p>
      ごちうサイクル - ご注文はサイクルですか？ -
    </p>
  </header>
  <div>
    <button
      type="button"
    >
      回転
    </button>
    <button
      type="button"
    >
      逆回転
    </button>
  </div>
</div>
`;
